pennylane/qchem/convert.py

```diff
@@ -354,3 +354,81 @@ def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e01
         return qml.dot(*_openfermion_to_pennylane(qubit_observable, wires=wires))
 
     return qml.Hamiltonian(*_openfermion_to_pennylane(qubit_observable, wires=wires))
+
+
+def _excitations(electrons, orbitals):
+    r"""Generate all possible single and double excitations from a Hartree-Fock reference state.
+
+    Args:
+        electrons (int): number of electrons
+        orbitals (int): number of spin orbitals
```

Comment:
we should be consistent in the usage of spin orbitals vs molecular orbitals. I agree that it can be used correctly if one remembers, but maybe we should just convert all mentions of orbitals to one type. For example, in the wf constructors like _ucisd_state relying on this functionality (and future constructors), the alpha and beta electron occupation vectors are constructed separately, so it is natural to work with molecular orbitals, not spin orbitals.

Or maybe we just say "orbitals" and let the user decide what it means for them.

Reply:
The input to this function is the number of electrons and the number of _spin_ orbitals, not molecular orbitals. Why we should change to orbitals?

Reply:
Inside the wavefunction constructor `_cisd_state`, the function `_excited_states` is called to generate excitations of either alpha electrons or beta electrons, but not both. If the number of spin orbitals is N, the relevant number of orbitals for this excitation of either alpha or beta sectors should be N/2 == number of molecular orbitals. 

I agree that it is just factors of two, but could be a bit confusing. Perhaps it should be changed on the `_cisd_state` side, e.g. we should use spin orbitals there too and just divide by two when calling `_excited_states`. Just thought I'd raise it for discussion.